test/RachisTests/DatabaseCluster/AtomicClusterReadWriteTests.cs

```diff
@@ -0,0 +1,281 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests.Server.Replication;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Operations.CompareExchange;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Smuggler;
+using Raven.Client.Exceptions;
+using Raven.Client.Util;
+using Raven.Server;
+using Raven.Tests.Core.Utils.Entities;
+using Sparrow.Server;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace RachisTests.DatabaseCluster
+{
+    public class AtomicClusterReadWriteTests : ReplicationTestBase
+    {
+        public AtomicClusterReadWriteTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        class TestObj
+        {
+            public string Id { get; set; }
+            public string Prop { get; set; }
+        }
+
+        [Fact]
+        public async Task ClusterWideTransaction_WhenStore_ShouldCreateCompareExchange()
+        {
+            var (nodes, leader) = await CreateRaftCluster(3);
+            using var store = GetDocumentStore(new Options {Server = leader, ReplicationFactor = nodes.Count});
+
+            var entity = new TestObj();
+            using (var session = store.OpenAsyncSession(new SessionOptions { TransactionMode = TransactionMode.ClusterWide }))
+            {
+                await session.StoreAsync(entity);
+                await session.SaveChangesAsync();
+            }
+
+            var result = await store.Operations.SendAsync(new GetCompareExchangeValuesOperation<User>(""));
```

Comment:
??

Reply:
Just checking a `CompareExchange` was created.
What does bother you here?
The empty string?